### PR TITLE
Update to new scalismo version

### PR DESCRIPTION
**Pull request by Andreas-Forster**
An initial preview of changes needed to update to new scalismo dependency and the complete migrating to scala 3. Yet not all warnings are fixed.

### CODE REVIEW

File: src/main/scala/scalismo/faces/sampling/face/proposals/SphericalHarmonicsLightProposals.scala [hunk 1]
```diff
@@ -251,7 +250,7 @@ object SphericalHarmonicsLightProposals {
           if (math.abs(diff.x - diff.y) + math.abs(diff.y - diff.z) < monochromaticityThreshold) { //if monochromatic change, then we could take any channel, we choose the first one.
             totProb += 3.0 * (-0.5 * math.pow(diff.x / sdev, 2) - 0.5 * math.log(2.0 * math.Pi) - math.log(sdev))
           } else {
-            return Double.NegativeInfinity //if non monochromatic change
+            returning(Double.NegativeInfinity) //if non monochromatic change
```
Comment (marcelluethi) [inline src/main/scala/scalismo/faces/sampling/face/proposals/SphericalHarmonicsLightProposals.scala]: what does returning mean? Shouldn't this just be `Double.NegativeInfinity`?

File: src/test/scala/scalismo/faces/momo/MoMoTests.scala [hunk 2]
```diff
@@ -162,7 +167,7 @@ class MoMoTests extends FacesTestSuite {
       norm(meanCoeffs.shape) + norm(meanCoeffs.color) should be < 0.01 * momoCoeffs.shape.length + 0.01 * momoCoeffs.color.length
     }
 
-    val f = File.createTempFile("momo", ".h5")
+    val f = File.createTempFile("momo", ".json")
```
Comment (marcelluethi) [inline src/test/scala/scalismo/faces/momo/MoMoTests.scala]: I usually call the files `.h5.json` but this works as well.

File: project/build.properties [hunk 3]
```diff
@@ -1 +1 @@
-sbt.version=1.3.5
+sbt.version=1.6.1
```
Comment (marcelluethi) [inline project/build.properties]: Maybe we should go directly to 1.8?

**Review by marcelluethi — COMMENTED** [review]
Thanks for starting this update. This is great. 
I added a few comments inline. 

What surprises me is that I cannot see a change in build.sbt?

**Comment by marcelluethi** [comment]
What happened to the integration tests? I see that no CI is running?

**Comment by Andreas-Forster** [comment]
You were right, the build.sbt was indeed missing. I corrected that now.

I do not yet know what happened to the CI. Maybe never setup correctly or it was not maintained and is now outdated.

File: src/main/scala/scalismo/faces/sampling/face/proposals/SphericalHarmonicsLightProposals.scala (hunk 1 above)
Comment (Andreas-Forster) [reply src/main/scala/scalismo/faces/sampling/face/proposals/SphericalHarmonicsLightProposals.scala]: There was a warning that returning from an nested anonymous function is deprecated since Scala 3. And as far as I understood, the returning is a possible library replacement from scala.util.control.NonLocalReturns.

File: src/test/scala/scalismo/faces/momo/MoMoTests.scala (hunk 2 above)
Comment (Andreas-Forster) [reply src/test/scala/scalismo/faces/momo/MoMoTests.scala]: That is a good idea - thx.

File: project/build.properties (hunk 3 above)
Comment (Andreas-Forster) [reply project/build.properties]: I don't mind - will update that.

**Comment by Andreas-Forster** [comment]
I added CI and fixed formatting to follow the style checks already used in Scalismo.

Further, I updated the file endings of models to be ".h5.json" in the tests and it uses now sbt 1.8.2 for building.

File: build.sbt [hunk 4]
```diff
@@ -15,28 +15,26 @@ lazy val root = (project in file("."))
...
+    scalaVersion  := "3.2.2",
+    scalacOptions ++= Seq("-deprecation", "-unchecked", "-feature"), // "-opt:l:method",
+    javacOptions ++= Seq("-source", "1.8", "-target", "1.8"),
     libraryDependencies  ++= Seq(
-      "ch.unibas.cs.gravis" %% "scalismo" % "0.90.0",
-      "ch.unibas.cs.gravis" % "scalismo-native-all" % "4.0.1",
-      "org.scalatest" %% "scalatest" % "3.0.8" % "test"
+      "ch.unibas.cs.gravis" %% "scalismo" % "0.92-SNAPSHOT",
+      "org.scalatest" %% "scalatest" % "3.2.15" % Test,
+      "io.spray" %%  "spray-json" % "1.3.6"
```
Comment (marcelluethi) [inline build.sbt]: Maybe we should think about replacing spray json with ujson. We use it only in two files and in scalismo we migrated to it recently. The less dependencies we are introducing the better. 
But it any case, this should be a separate PR, which I also would be willing to do.

**Review by marcelluethi — APPROVED** [review]
Thanks a lot for all the changes. This looks very good.

File: build.sbt [hunk 5]
```diff
@@ -6,37 +6,40 @@ val scalismoPlatform = {
...
-      CrossVersion.partialVersion(scalaVersion.value) match {
-        case Some((2, n)) if n >= 13 => sourceDir / "scala-2.13+"
-        case _                       => sourceDir / "scala-2.13-"
-      }
-    }
+    scalaVersion := "3.2.2",
+    scalacOptions ++= Seq("-deprecation", "-unchecked", "-feature"), // "-opt:l:method",
+    javacOptions ++= Seq("-source", "1.8", "-target", "1.8"),
+    libraryDependencies ++= Seq(
+      "ch.unibas.cs.gravis" %% "scalismo" % "0.92-SNAPSHOT",
```
Comment (marcelluethi) [inline build.sbt]: This can now be replaced with 0.92-RC1.
